rfcs/054-id-resolver/README.md

```diff
@@ -0,0 +1,155 @@
+By resolving those issues before the ID Minter, those Concepts will actually be the same Concept when they reach the
+Concepts Pipeline, rather than multiple Concepts that need to be understood as synonymous.
+
+The proposed stage here operates on Authoritative Concept Identifiers, ensuring that the resulting Catalogue Concepts 
+are correct and consistent before new Canonical Identifiers are minted for them.
+
+## How?
+
+### Concepts Pipeline/API
+The Concepts Pipeline Ingestor(s) should, where available, store the official types (maybe translated to our types) 
+and compositions of each Concept in the authoritative-concepts index.
+
+The Concepts API should have a service that can be called with one or more authoritative identifiers, and returns that information
+from the authoritative index.  This may be a separate private/internal API instance.
+
+The Sierra Transformer should stop guessing whether to use LCNames or LCSubjects, and instead use a new scheme, "LC".
+
+The Catalogue Pipeline should have a new stage between the transformers and the id minter, the new stage:
+
+* extracts identifiers in the same way the id minter does, and requests the official type/composition for any that are in a scheme supported by the API.
+* replaces the type and identifiertype with the "right" value from the API
+* replaces the identifier values of any constituent concepts that have label-derived identifiers, with the official ones (based on similarity of label).
+* rejects and/or warns about documents containing any wrong identifiers (initially, just not known in that scheme, later, consider some comparison between labels).
+
+Everything downstream will behave exactly as it currently does.
+
+## Why a new stage?
+
+This must be done after the identifiers have been found by a transformer and
+before the ID Minter mints canonical identifiers in the id minter database.
+
+Therefore, other options for doing this work could include adding to the ID Minter or the transformers.
+
+It does not belong in the transformers.  Although this problem mostly manifests in Sierra data, it could come from anywhere.
+For example, there are examples coming from TEI data.  The transformers work best by being self-contained, and only transform
+the input given into a Work, rather than performing lookups elsewhere.  So it is best not to burden transformers with this.
+
+It may belong in the ID Minter.  There is some duplication between this task and the id minter, in the extraction of source
+identifiers from the input Work. The only reasons for keeping it separate are - 
+
+* better observability (weak, it's probably just as clear what has happened if resolution and minting are in the same stage)
+* the minter is fairly critical and may be best left alone (also weak, it's well covered in tests) 
+
+## Current Problem Manifestations
+
+### The Person/Agent (Maimonides) problem
```

Comment:
I might be misunderstanding - is this a data problem (ie to be resolved with collections info) or is it fundamental to the problem ie unavoidable?

Reply:
This is a good point.  I had thought of it as needing a technological solution, due to the scale of all of this.  However, I should have a proper look at the scale of this particular one, and seek counsel from an information professional.